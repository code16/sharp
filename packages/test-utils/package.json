{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}