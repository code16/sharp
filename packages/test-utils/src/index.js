export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}