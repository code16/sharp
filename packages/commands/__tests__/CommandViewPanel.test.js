import { mount } from '@vue/test-utils';
import ViewPanel from '../src/components/CommandViewPanel.vue';

describe('view-panel', ()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = mount(ViewPanel, {
            directives: {
                srcdoc() {}
            }
        });
    });

    test('mount ViewPanel', ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount "visible with content" ViewPanel', ()=>{
        wrapper.setProps({ content: 'SOME HTML MARKUP' });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('emit "change" to false on clicked background glasspane', ()=>{
        wrapper.setProps({ show: true });
        let glasspane = wrapper.find('.SharpViewPanel__glasspane');
        glasspane.trigger('click');
        expect(wrapper.emitted().close).toHaveLength(1);
    });

});