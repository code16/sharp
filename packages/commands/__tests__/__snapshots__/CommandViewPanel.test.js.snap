// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view-panel mount "visible with content" ViewPanel 1`] = `
<div>
	<div class="SharpViewPanel__glasspane" style=""></div>
	<div class="SharpViewPanel" enter-class="SharpViewPanel--collapsed" enter-active-class="SharpViewPanel--expanding" enter-to-class="SharpViewPanel--expanded" leave-class="SharpViewPanel--expanded" leave-active-class="SharpViewPanel--collapsing"
	    leave-to-class="SharpViewPanel--collapsed">
		<iframe src="about:blank" sandbox="allow-forms allow-scripts allow-same-origin allow-popups"></iframe>
	</div>
</div>
`;

exports[`view-panel mount ViewPanel 1`] = `
<div>
	<div class="SharpViewPanel__glasspane" style="display: none;"></div>
	<!---->
</div>
`;
